server/src-lib/Hasura/RQL/DDL/EventTrigger.hs

```diff
@@ -140,45 +137,52 @@ addEventTriggerToCatalog
   -> [PGColInfo]
   -> Bool
   -> EventTriggerConf
-  -> Q.TxE QErr TriggerId
+  -> Q.TxE QErr ()
 addEventTriggerToCatalog qt allCols strfyNum etc = do
-  ids <- map runIdentity <$> Q.listQE defaultTxErrorHandler
+  Q.unitQE defaultTxErrorHandler
          [Q.sql|
            INSERT into hdb_catalog.event_triggers
                        (name, type, schema_name, table_name, configuration)
            VALUES ($1, 'table', $2, $3, $4)
-           RETURNING id
-               |] (name, sn, tn, Q.AltJ $ toJSON etc) True
+         |] (name, sn, tn, Q.AltJ $ toJSON etc) True
 
-  trid <- getTrid ids
-  mkTriggerQ trid name qt allCols strfyNum opsdef
-  return trid
+  mkTriggerQ name qt allCols strfyNum opsdef
   where
     QualifiedObject sn tn = qt
     (EventTriggerConf name opsdef _ _ _ _) = etc
-    getTrid []    = throw500 "could not create event-trigger"
-    getTrid (x:_) = return x
 
-delEventTriggerFromCatalog :: TriggerName -> Q.TxE QErr ()
-delEventTriggerFromCatalog trn = do
+delEventTriggerFromCatalog :: TriggerName -> Bool -> Q.TxE QErr ()
+delEventTriggerFromCatalog trn cascade = do
   Q.unitQE defaultTxErrorHandler [Q.sql|
            DELETE FROM
                   hdb_catalog.event_triggers
            WHERE name = $1
                 |] (Identity trn) True
   delTriggerQ trn
+  bool (return ()) (deleteAllEventsForTrigger trn) cascade
```

Comment:
is there a constant function for `returnNil` or something?

Reply:
Use `when` function.
```haskell
when cascade $ deleteAllEventsForTrigger trn
```